nxc/protocols/ssh.py

```diff
@@ -63,46 +70,137 @@ def create_conn_obj(self):
         except socket.error:
             return False
 
-    def client_close(self):
-        self.conn.close()
-
     def check_if_admin(self):
+        self.admin_privs = False
+
+        if self.args.sudo_check:
+            self.check_if_admin_sudo()
+            return
+
         # we could add in another method to check by piping in the password to sudo
         # but that might be too much of an opsec concern - maybe add in a flag to do more checks?
-        stdin, stdout, stderr = self.conn.exec_command("id")
-        if stdout.read().decode("utf-8").find("uid=0(root)") != -1:
-            self.logger.info(f"Determined user is root via `id` command")
-            self.admin_privs = True
-            return True
-        stdin, stdout, stderr = self.conn.exec_command("sudo -ln | grep 'NOPASSWD: ALL'")
-        if stdout.read().decode("utf-8").find("NOPASSWD: ALL") != -1:
-            self.logger.info(f"Determined user is root via `sudo -ln` command")
-            self.admin_privs = True
-            return True
+        self.logger.info(f"Determined user is root via `id && sudo -ln` command")
+        stdin, stdout, stderr = self.conn.exec_command("id && sudo -ln 2>&1")
```

Comment:
I don't think we should use && to check for this - we should run it one by one like previously.

Reply:
Yes

Reply:
Ha, I think ";" is better